mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/dao/DaoSelectMethodGenerator.java

```diff
@@ -157,22 +193,66 @@ private void generateSelectRequest(
       MethodSpec.Builder methodBuilder,
       String requestName,
       String helperFieldName,
-      int parameterSize) {
-    String customWhereClause = methodElement.getAnnotation(Select.class).customWhereClause();
+      int numberOfPrimaryKeyPartsInWhereClause) {
+    Select annotation = methodElement.getAnnotation(Select.class);
+    String customWhereClause = annotation.customWhereClause();
     if (customWhereClause.isEmpty()) {
-      methodBuilder.addStatement(
-          "$T $L = $L.selectByPrimaryKeyParts($L).build()",
+      methodBuilder.addCode(
+          "$[$T $L = $L.selectByPrimaryKeyParts($L)",
           SimpleStatement.class,
           requestName,
           helperFieldName,
-          parameterSize);
+          numberOfPrimaryKeyPartsInWhereClause);
     } else {
-      methodBuilder.addStatement(
-          "$T $L = $L.selectStart().whereRaw($S).build()",
+      methodBuilder.addCode(
+          "$[$T $L = $L.selectStart().whereRaw($S)",
           SimpleStatement.class,
           requestName,
           helperFieldName,
           customWhereClause);
     }
+    maybeAddSimpleClause(annotation.limit(), Integer::parseInt, "limit", "limit", methodBuilder);
+    maybeAddSimpleClause(
+        annotation.perPartitionLimit(),
+        Integer::parseInt,
+        "perPartitionLimit",
+        "perPartitionLimit",
+        methodBuilder);
+    for (String orderingSpec : annotation.orderBy()) {
+      addOrdering(orderingSpec, methodBuilder);
+    }
+    for (String groupByColumn : annotation.groupBy()) {
+      methodBuilder.addCode(".groupBy($S)", groupByColumn);
+    }
+    if (annotation.allowFiltering()) {
+      methodBuilder.addCode(".allowFiltering()");
+    }
+    methodBuilder.addCode(".build();$]\n");
   }
+
+  private void addOrdering(String orderingSpec, MethodSpec.Builder methodBuilder) {
+    List<String> tokens = ON_SPACES.splitToList(orderingSpec);
+    ClusteringOrder clusteringOrder;
+    if (tokens.size() != 2 || (clusteringOrder = parseClusteringOrder(tokens.get(1))) == null) {
```

Comment:
are we supporting the `orderBy` to have multiple statements? I.e:
```cc DESC aa DESC```
If so why we are validating only for tokens.size() != 2? Should we allow tokens to have `size() % 2 == 0`?

Reply:
The annotation value is an array, you would have one element per column:
```java
@Select(orderBy = { "cc DESC", "aa DESC" })
```
I will amend the javadocs and manual to explicitly include an example.